q2_types/feature_data/_format.py

```diff
@@ -131,23 +134,44 @@ def sniff(self):
 
 
 class DNAFASTAFormat(model.TextFileFormat):
-    def sniff(self):
-        filepath = str(self)
-        sniffer = skbio.io.io_registry.get_sniffer('fasta')
-        if sniffer(filepath)[0]:
-            generator = skbio.io.read(filepath, constructor=skbio.DNA,
-                                      format='fasta', verify=False)
-            try:
-                for seq, _ in zip(generator, range(5)):
-                    pass
-                return True
-            # ValueError raised by skbio if there are invalid DNA chars.
-            except ValueError:
-                pass
+    def _validate_lines(self, max_lines):
+        FASTADNAValidator = re.compile(r'[ACGTURYKMSWBDHVN\-\.]+\n?')
```

Comment:
Was thinking about this, and I think we actually don't want to support `-` and `.` as those are part of an aligned fasta file, which we call `AlignedDNAFastaFormat`. The good news is I think we could just subclass this one to implement `AlignedDNAFastaFormat`. All of the logic here should work, with the exception of permitting gap characters (`.` and `-`). 

That means I think you could make `FASTADNAValidator` a class property and just override it in the subclass to include the dash and dot.

That said, this is kind of out of scope of the original PR, so I'm fine with not doing all that and just removing the gap characters from this regex.

Reply:
I think for now I'll just remove the gap characters from the current regex and something along these lines can be done when we validate the other fasta formats